File: server/src/map-provider.js
export function getMapProviderConfig() {
  const provider = (process.env.MAP_PROVIDER || 'auto').toLowerCase();

  return {
    provider,
    googleConfigured: Boolean(process.env.GOOGLE_MAPS_API_KEY),
    neshanConfigured: Boolean(process.env.NESHAN_API_KEY),
    baladConfigured: Boolean(process.env.BALAD_API_KEY),

    allowedMapProviders: ['google', 'neshan', 'balad', 'osm'],
    allowedNavigationProviders: ['waze', 'google', 'neshan', 'balad'],

    fallback: 'osm',
  };
}

/*
  MTcar provider policy

  In-app maps:
  - Google: official Google Maps SDK/API key required.
  - Neshan: enable only with valid official SDK/API credentials.
  - Balad: enable only when official documented developer access is available.
  - OSM: development/fallback provider.

  Navigation:
  - Waze is treated as an external navigation target/deep-link, not as an
    embeddable base-map layer.
  - Google/Neshan/Balad can also be navigation targets when supported.

  Do not hard-code undocumented map tile URLs.
*/

File: mobile/lib/maps/map_provider.dart
enum MapProviderType {
  auto,
  google,
  neshan,
  balad,
  osm,
}

enum NavigationProviderType {
  waze,
  google,
  neshan,
  balad,
}

enum MapVisualStyle {
  standard,
  satellite,
  dark,
}

class MapProviderAvailability {
  final bool googleConfigured;
  final bool neshanConfigured;
  final bool baladConfigured;

  const MapProviderAvailability({
    required this.googleConfigured,
    required this.neshanConfigured,
    required this.baladConfigured,
  });

  bool isAvailable(MapProviderType type) {
    switch (type) {
      case MapProviderType.auto:
      case MapProviderType.osm:
        return true;
      case MapProviderType.google:
        return googleConfigured;
      case MapProviderType.neshan:
        return neshanConfigured;
      case MapProviderType.balad:
        return baladConfigured;
    }
  }

  MapProviderType resolve(MapProviderType preferred) {
    if (preferred != MapProviderType.auto && isAvailable(preferred)) {
      return preferred;
    }

    if (neshanConfigured) return MapProviderType.neshan;
    if (googleConfigured) return MapProviderType.google;
    if (baladConfigured) return MapProviderType.balad;
    return MapProviderType.osm;
  }
}

/*
  Important:
  - Google must use the official Google Maps SDK/API key.
  - Neshan/Balad must only use official documented SDK/API access.
  - Waze is a navigation target, not an embedded base-map provider.
  - OSM remains the safe development fallback.
*/
